### PR TITLE
Migrate from setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by pefoley2**
Based on https://setuptools.pypa.io/en/latest/userguide

## Proposed changes

Migrate configuration from setup.py to pyproject.toml, but leave a stub setup.py to avoid breaking setup.py test and friends.

## Brief description of rationale

Setuptools is discouraging direct usage of setup.py https://setuptools.pypa.io/en/latest/userguide/quickstart.html#transitioning-from-setup-py-to-setup-cfg

### CODE REVIEW

File: Ion.egg-info/SOURCES.txt
Comment (alanzhu0) [inline Ion.egg-info/SOURCES.txt:1]: Please leave secret.py in SOURCES.txt

File: Ion.egg-info/top_level.txt
Comment (alanzhu0) [inline Ion.egg-info/top_level.txt:1]: I don't think this should be removed?

File: intranet/apps/notifications/emails.py
Comment (alanzhu0) [inline intranet/apps/notifications/emails.py:1]: Please exclude this file from the commit.

File: requirements.txt
Comment (alanzhu0) [inline requirements.txt:1]: What was the change here? Is it just Git being weird?

**Review by alanzhu0 — CHANGES_REQUESTED** [review]
Hey there! Thanks so much for your contribution. 

Your setup.py -> pyproject.toml changes look great. The only thing is there are a few files not directly connected to that change in the commit - I will point those out in the file comments. Could you remove those files from the commit?

Thanks!

**Comment by NotFish232** [comment]
Closed in favor of #1648

**Pull request closed without merging**